### PR TITLE
Add test for image and video CNNs

### PR DESCRIPTION
**Pull request by XianyuanLiu**
### Description
Add test for video cnns and image cnns.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.

### CODE REVIEW

File: tests/embed/test_video_feature_extractor.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+@pytest.mark.parametrize("attention", ATTENTION)
+@pytest.mark.parametrize("num_classes", NUM_CLASSES)
+def test_get_video_feat_extractor(model_name, image_modality, attention, num_classes):
+    feature_network, class_feature_dim, domain_feature_dim = get_video_feat_extractor(
+        model_name, image_modality, attention, num_classes
+    )
+
+    assert isinstance(feature_network, dict)
+
+    assert isinstance(feature_network, dict)
```
Comment (haipinglu) [inline tests/embed/test_video_feature_extractor.py:23]: Duplication with line 21, remove one of them?

File: tests/embed/test_video_i3d_r3d.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+from kale.embed.video_res3d import BasicBlock, BasicStem, Bottleneck, Conv3DSimple, VideoResNet
+from kale.embed.video_se_i3d import SEInceptionI3DFlow, SEInceptionI3DRGB
+from kale.embed.video_se_res3d import se_r3d_18_flow, se_r3d_18_rgb
+from kale.utils.seed import set_seed
+
+# the default input shape is batch_size * num_channel * frame_per_segment * height * weight
+INPUT_BATCH_RGB = torch.randn(2, 3, 16, 112, 112)
+INPUT_BATCH_FLOW = torch.randn(2, 2, 8, 112, 112)
+SE_LAYERS = ["SELayerC", "SELayerT", "SELayerCoC", "SELayerMC", "SELayerMAC", "SELayerCT", "SELayerTC"]
+set_seed(36)
```
Comment (haipinglu) [inline tests/embed/test_video_i3d_r3d.py:16]: I know that this won't fail the tests but it should be better to always set seed before using random functions, i.e., putting it before line 13 will be better, isn't it?

File: tests/predict/test_class_domain_nets.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    ClassNetVideo,
+    ClassNetVideoConv,
+    DomainNetSmallImage,
+    DomainNetVideo,
+    SoftmaxNet,
+)
+
+BATCH_SIZE = 2
+INPUT_BATCH = torch.randn(BATCH_SIZE, 128)
+INPUT_BATCH_LOGITS = torch.randn(BATCH_SIZE, 1024, 1, 1, 1)
```
Comment (haipinglu) [inline tests/predict/test_class_domain_nets.py:15]: I could guess what it is for but adding some light comment about what line 15 is will help users or new comers. This code not only serves the purpose of test coverage, but also can teach others how to write proper tests. Thanks.

**Review by haipinglu — CHANGES_REQUESTED** [review]
Amazing efforts to have the coverage _doubled_, from 25% to 51%!!!!!!

Just some minor comments to address before merging.

**Comment by XianyuanLiu** [comment]
> Amazing efforts to have the coverage _doubled_, from 25% to 51%!!!!!!
> 
> Just some minor comments to address before merging.

Updated.
Thanks! So happy last night! I didn't believe it. I thought something must be wrong and checked many times! We are closer to our goal now!

File: tests/predict/test_class_domain_nets.py (hunk 3 above)
Comment (haipinglu) [reply tests/predict/test_class_domain_nets.py:20]: @XianyuanLiu I should make it clearer: could you explain a bit `INPUT_BATCH_LOGITS` vs `INPUT_BATCH`? Is `INPUT_BATCH_LOGITS` some synthetic output from previous layers?

Or provide some references about the `_LOGITS`. It is not as obvious as `INPUT_BATCH`.

Comment (haipinglu) [reply tests/predict/test_class_domain_nets.py:20]: > @XianyuanLiu I should make it clearer: could you explain a bit `INPUT_BATCH_LOGITS` vs `INPUT_BATCH`? Is `INPUT_BATCH_LOGITS` some synthetic output from previous layers (i.e. output of the I3D)?
> 
> Or provide some references about the `_LOGITS`. It is not as obvious as `INPUT_BATCH`.

Or maybe changing `_LOGITS` to `_I3D`?

**Review by haipinglu — APPROVED** [review]

**Pull request merged**